src/Text/Pandoc/App/Opt.hs

```diff
@@ -130,7 +131,8 @@ data Opt = Opt
     , optPdfEngine             :: Maybe String -- ^ Program to use for latex/html -> pdf
     , optPdfEngineOpts         :: [String]   -- ^ Flags to pass to the engine
     , optSlideLevel            :: Maybe Int  -- ^ Header level that creates slides
-    , optSetextHeaders         :: Bool       -- ^ Use atx headers for markdown level 1-2
+    -- | Heading style for markdown level 1-2
+    , optMarkdownHeadingFormat :: MarkdownHeadingFormat
```

Comment:
Instead of the comment on line 134, put a comment on line 135 using `-- ^ ` as above.

Reply:
Done. I was thinking about the 80 char line width guideline. Also I was trying to avoid modifying the spacing before the '::'s.

Reply:
I'm sorry to reverse myself on this, but on reflection, I think that it's a bad idea to break the public API for aesthetic reasons. Let's just keep `optSetextHeaders` and use that.  Everything else can stay the same, and then we can integrate this change without a breaking API change.

Also, is it possible to rebase your changes on master so that they are easier to integrate, and possibly collapse them into a small number of commits?

Reply:
> I think that it's a bad idea to break the public API for aesthetic reasons. Let's just keep optSetextHeaders and use that.

I'm not sure I understand what you mean.

In this patch
1. `atx-headings` are the new default markdown headings.
2. `--atx-headers` is deprecated.
3. `--markdown-headings=setext|atx` is the new way of controlling the markdown headings.

Did you mean to revert any of those? Or you meant keeping those but implement them using optSetextHeaders? I think you mean the latter.

> Also, is it possible to rebase your changes on master so that they are easier to integrate, and possibly collapse them 
> into a small number of commits?

I'll see if I can to collapse some commits on test cases.

Reply:
I guess what I had in mind was avoiding the switch from `optSetextHeaders` to `optMarkdownHeadingFormat`.  That change would require a bump to 2.12.
I'm actually sort of conflicted on this.  I'm not ready to go to 2.12 yet, but otherwise we could merge this change.  On the other hand, the change does improve clarity.  (There may also be ramifications for the JSON and YAML instances that I'm not thinking of.)

Reply:
In any case, collapsing to a small number of commits makes it easier to rebase in the future.

Reply:
I can't remember whether your changes included also a change to `writerSetextHeaders` in `WriterOptions`.

Seems to me that if we're going to make a big API change, we might consider changing that as well as `optSetextHeaders`.

But maybe the principle of avoiding API changes except when necessary favors just leaving these types as they are?

Reply:
Very well, I'll keep `optSetextHeaders` and make `----markdown-headings=setext|atx` modify it accordingly.